src/list/components/item-fields.js

```diff
@@ -121,18 +124,36 @@ export class EditItemFields extends React.Component {
 
     return (
       <div className={styles.itemFields}>
-        <label>
-          <Localized id="item-fields-origin">
-            <LabelText>oRIGIn</LabelText>
-          </Localized>
-          <Localized id="item-fields-origin-input" attrs={{placeholder: true}}>
-            <Input className={styles.input} type="text"
-                   disabled={!!fields.itemId}
-                   placeholder="hTTps://wWw.eXAMPLe.cOm"
-                   {...controlledProps("origin")}
-                   ref={(element) => this._firstField = element} />
-          </Localized>
-        </label>
+        <div className={styles.inputWrap}>
+          <label>
+            <Localized id="item-fields-origin">
+              <LabelText>oRIGIn</LabelText>
+            </Localized>
+            <Localized id="item-fields-origin-input" attrs={{placeholder: true}}>
+              <Input className={styles.originInput} type="text"
+                     disabled={!!fields.itemId}
+                     pattern={originPattern}
```

Comment:
If you want to just use html5 input validation, I'd use `type=url` here, and skip the `pattern`. That should get us everything we need. You could add a prop like `validationAttrs` to pass `type` or `pattern` or both down as needed.